lib/resources.js

```diff
@@ -775,21 +790,18 @@ const buildElasticQuery = function (params) {
 
   if (params.q) {
     // Merge keyword-specific ES query into the query we're building:
-    query.bool = buildElasticQueryForKeywords(params).bool
+    query.bool = { must: [ buildElasticQueryForKeywords(params) ] }
   }
 
   const advancedParamQueries = ADVANCED_SEARCH_PARAMS
     .filter((param) => params[param])
     .map((param) =>
-      buildElasticQueryForKeywords({ q: params[param], search_scope: param }).bool.should
+      buildElasticQueryForKeywords({ q: params[param], search_scope: param })
     )
 
   if (advancedParamQueries.length) {
     query.bool = query.bool || {}
-    query.bool.must = (query.bool.should || []).concat(advancedParamQueries.reduce((acc, el) =>
-      acc.concat(el)
-    ))
```

Comment:
`advancedParamQueries` is an array of arrays (of clauses previously bound in `should`s). I think by flattening `advancedParamQueries`, you're potentially placing clauses that used to be bound by a `should` (boolean OR) into a `must` (boolean AND)